tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+
+							err := fmt.Errorf("Something must have gone wrong. You can try restarting the test")
+							log.FailOnError(err, "Unexpected Status")
+							return "", false, err
+						}
+
+						return "", true, fmt.Errorf("waiting for restore status to be PartialSuccess; got [%s]",
+							restoreResponseStatus.GetStatus())
+					}
+					_, err = task.DoRetryWithTimeout(restorePartialSuccessCheck, 10*time.Minute, 30*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restore 2 (%s) is PartialSuccess", laterRestoreName))
+
+					// Validation of Restore
+					restoreInspectRequest = &api.RestoreInspectRequest{
+						Name:  laterRestoreName,
+						OrgId: orgID,
+					}
+
+					backupDriver := Inst().Backup
+					resp, err := backupDriver.InspectRestore(ctx, restoreInspectRequest)
+					log.FailOnError(err, "Failed to inspect restore: %s", laterRestoreName)
+
+					log.InfoD("Switching to destination context")
+					err = SetDestinationKubeConfig()
+					log.FailOnError(err, "Failed to switch to context to destination cluster")
+
+					rstCtx, err := ValidateRestore(resp, backupContexts[i], namespaceMapping)
+
+					log.InfoD("Switching to default context")
+					err1 := SetClusterContext("")
+					log.FailOnError(err1, "Failed to SetClusterContext to default cluster")
+
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restore 2 (%s) Validation", laterRestoreName))
+
+					restoreLaterContexts = append(restoreLaterContexts, rstCtx)
+				})
+
+				Step("Verifying status of Initial Restore", func() {
+					log.InfoD("Step: Verifying status of Initial Restore")
+
+					restoreInspectRequest := &api.RestoreInspectRequest{
+						Name:  initialRestoreName,
+						OrgId: orgID,
+					}
+					restoreSuccessCheck := func() (interface{}, bool, error) {
+						resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+						restoreResponseStatus := resp.GetRestore().GetStatus()
+						log.FailOnError(err, "Failed getting restore status for - %s", initialRestoreName)
+
+						if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Success {
+							log.Infof("Restore status - %s", restoreResponseStatus)
+							log.InfoD("Status of %s - [%s]",
+								initialRestoreName, restoreResponseStatus.GetStatus())
+							log.InfoD("Condition fulfilled. Proceeding to end test")
+							return "", false, nil
+						} else if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess {
+							log.Infof("Restore status - %s", restoreResponseStatus)
+							log.InfoD("Status of %s - [%s]",
+								initialRestoreName, restoreResponseStatus.GetStatus())
+
+							err := fmt.Errorf("Status of Restore 1 (%s) is PartialSuccess. This should not happen. This MAY have happened if the destination cluster hadn't been cleaned completely (remanant cluster-level resources). The OTHER situation which may have caused this is explained in the comment of JIRA ticket PA-614", initialRestoreName)
```

Comment:
Keep the error message short and crisp

Reply:
I have significantly shortened the error messages and even cleaned and consolidated logging, but I still left in _really short and crisp help messages_. After going through them, if you still feel that it is way too much OR unnecessary OR confusing in nature OR some other reason, I'll restrict it to error-only messages. 😄

full reply: https://github.com/portworx/torpedo/pull/1182#discussion_r1134952153